src/doc/dynamic-throttle.txt

```diff
@@ -0,0 +1,129 @@
+     induce delays to the queueing threads if the ratio of current outstanding 
+     bytes|ops and filestore_queue_max_(bytes|ops) is more than 
+     filestore_queue_low_threshhold. The throttles will block io once the 
+     outstanding bytes|ops reaches max value (filestore_queue_max_(bytes|ops)).
+
+     User configurable config option for adjusting delay is the following.
+  
+        filestore_queue_low_threshhold:
+          Valid values should be between 0-1 and shouldn't be > *_high_threshold.
+
+        filestore_queue_high_threshhold:
+          Valid values should be between 0-1 and shouldn't be < *_low_threshold.
+
+        filestore_expected_throughput_ops:
+          Shouldn't be less than or equal to 0. 
+
+	filestore_expected_throughput_bytes:
+          Shouldn't be less than or equal to 0.
+ 
+        filestore_queue_high_delay_multiple:
+	  Shouldn't be less than 0 and shouldn't be > *_max_delay_multiple.
+          
+	filestore_queue_max_delay_multiple:
+	  Shouldn't be less than 0 and shouldn't be < *_high_delay_multiple
+
+  2. journal usage throttle is implemented to gradually slow down queue_transactions 
+     callers as the journal fills up. We don't want journal to become full as this
+     will again induce spiky behavior. The configs work very similarly to the
+     Filestore op worker queue throttle.
+
+        journal_throttle_low_threshhold
+	journal_throttle_high_threshhold
+	filestore_expected_throughput_ops
+	filestore_expected_throughput_bytes
+	journal_throttle_high_multiple
+	journal_throttle_max_multiple
+
+This scheme will not be inducing any delay between [0, low_threshold].
+In [low_threshhold, high_threshhold), delays should be injected based on a line
+from 0 at low_threshold to high_multiple * (1/expected_throughput) at high_threshhold.
+In [high_threshhold, 1), we want delays injected based on a line from
+(high_multiple * (1/expected_throughput)) at high_threshhold to 
+(high_multiple * (1/expected_throughput)) + (max_multiple * (1/expected_throughput)) 
+at 1.
+Now setting *_high_multiple and *_max_multiple to 0 should give us similar effect
+like old throttle. Setting filestore_queue_max_ops and filestore_queue_max_bytes
+to zero should disable the entire backoff throttle.
+          
+-------------------------
+HOW TO PICK PROPER VALUE ?
+-------------------------
+
+General guideline is the following.
+
+  filestore_queue_max_bytes and filestore_queue_max_ops:
+  -----------------------------------------------------
+
+  This directly depends on how filestore op_wq will be growing in the memory. So,
+  user needs to calculate how much memory he can give to each OSD. Bigger value
+  meaning current/max ratio will be smaller and throttle will be relaxed a bit.
+  This could help in case of 100% write but in case of mixed read/write this can
+  induce more latencies for reads on the objects those are not yet been applied 
+  i.e yet to reach in it's final filesystem location.
```

Comment:
You should mention that once you reach max throughput, increasing these further only increases memory usage and latency (bad).